be/src/exprs/function_call_expr.cpp

```diff
@@ -173,4 +176,64 @@ StatusOr<ColumnPtr> VectorizedFunctionCallExpr::evaluate_checked(starrocks::Expr
     return result;
 }
 
+bool VectorizedFunctionCallExpr::ngram_bloom_filter(ExprContext* context, const BloomFilter* bf,
+                                                    size_t index_gram_num) const {
+    FunctionContext* fn_ctx = context->fn_context(_fn_context_index);
+    std::vector<Slice>& ngram_set = fn_ctx->get_ngram_set();
+
+    const auto& gram_num_column = fn_ctx->get_constant_column(2);
+    // case like ngram_search(col,"needle", 5) when col has a 4gram bloom filter, don't use this index
```

Comment:
5 in ngram_search(col,"needle", 5)  is typo? accoding to following words "when col has a 4gram bloom filter...",
so index_gram_num is the number of grams in storage engine?  predicate_gram_num is specified in users' sql?
It seems that these two params have not to be the same? for an example:  index_gram_num <= predicate_gram_num

Reply:
yes, it's not a typo. your description is correct, when create index, user need to offer gram_num(if not, default is 4 given by FE). when use ngram_search, user need to offer another parameter to specify the gram num of this function.  if these two gram num(index_gram_num and predicate_gram_num) is not the same, we can't use this index to filter date for this ngram_search function. Otherwise the result may be wrong